tests/naive.py

```diff
@@ -2080,10 +2080,12 @@ def find_matches(D, excl_zone, max_distance, max_matches=None):
     return np.array(result[:max_matches], dtype=object)
 
 
-def isconstant_func_stddev_threshold(a, w, quantile_threshold=0):
+def isconstant_func_stddev_threshold(a, w, quantile_threshold=0, stddev_threshold=None):
     sliding_stddev = rolling_nanstd(a, w)
-    if quantile_threshold == 0:  # pragma: no cover
-        return sliding_stddev == 0
-    else:
-        stddev_threshold = np.quantile(sliding_stddev, quantile_threshold)
-        return sliding_stddev <= stddev_threshold
+    if stddev_threshold is None:
```

Comment:
```
if stddev_threshold is None:
    stddev_threshold = np.quantile(sliding_stddev, quantile_threshold)
    if quantile_threshold == 0:
        stddev_threshold = 0
```

I didn't love the `else` as it was a bit harder to read due to it being nested

Reply:
@seanlaw 
I totally agree with you. What do you suggest @seanlaw ? I can go with what you initially suggested and `return` an ouput `if stddev_threshold is not None` 

```
rolling_stddev = rolling_nanstd(a, w)
if stddev_threshold is not None:
     return  rolling_stddev <  stddev_threshold

# else, meaning `stddev_threshold` is None
 if quantile_threshold == 0:
       stddev_threshold = 0  # reset to 0 
else:
       stddev_threshold = np.quantile(sliding_stddev, quantile_threshold)
```

I personally think we can replace the if/else block with this:

```
# else, meaning `stddev_threshold` is None
 stddev_threshold = np.quantile(sliding_stddev, quantile_threshold)
 if quantile_threshold == 0:
       stddev_threshold = 0  # replce computed `stddev_threshold` with 0
```

Reply:
Hmm, is there a problem with what I proposed above?

```
rolling_stddev = rolling_nanstd(a, w)

if stddev_threshold is None:
    stddev_threshold = np.quantile(rolling_stddev, quantile_threshold)
    if quantile_threshold == 0:
        stddev_threshold = 0

return rolling_stddev < stddev_threshold
```

Am I overlooking some logic here? It seems reasonably clear and okay to keep it all in my head. What I also don't love is returning a value in the middle of a function rather than at the end.

Reply:
I thought you were just quoting a part of my code and you just asked me to find a way to avoid nested if. I now noticed that the code in your initial comment was actually your suggestion :) All good! 

I knew about your concern about having return in the middle of the function but I thought you may allow this time :) I will revise the code accordingly.